src/main/js/html.js

```diff
@@ -302,9 +284,33 @@
                 context.bpd = "rl";
 
             }
+ 
+        } else if (isd_element.kind === "p" && context.bpd === "tb") {
+
+            var pdir = isd_element.styleAttrs[imscStyles.byName.direction.qname];
 
+            context.ipd = pdir === "ltr" ? "lr" : "rl"; 
```

Comment:
I'm not sure I understand why this change is necessary. Can you create an example?

Reply:
@cconcolato Even though it is not entirely clear in TTML2, `tts:direction`, when specified on `p`, is intended to override `tts:writingMode` (at least when ipd is horizontal), otherwise the following test does not work: https://github.com/w3c/imsc-tests/blob/master/imsc1/ttml/direction/Direction003.ttml . It is not clear what is the intended effect of `tts:direction` specified on `p` when the ipd is vertical -- in the case of CSS, this overrides the ipd vertically, where `ltr` means `tb` and `rtl` means `bt`.

Reply:
See https://github.com/w3c/ttml2/issues/1211